### PR TITLE
Refs #27301 - instead setup call let for usergroup audit tests

### PR DESCRIPTION
**Pull request by kgaikwad**
+ adding @tbrisker into the loop.

### CODE REVIEW

**Comment by kgaikwad** [comment]
[test foreman]

**Comment by kgaikwad** [comment]
[test foreman]
As these test failures are random, to make sure running test suite multiple times.

**Comment by ekohl** [comment]
@kgaikwad the test order is random, but you can provide a seed when you manually run them. That's probably the best way to reproduce it. Look for `--seed ...` in the command that ran it. Then you can pass `SEED=...` to reproduce.

File: test/models/usergroup_test.rb [hunk 1]
```diff
@@ -314,82 +314,88 @@ def populate_usergroups
...
-        @usergroup.usergroup_ids = [@child_usergroup.id]
-        @usergroup.save
-      end
+      let (:usergroup) { FactoryBot.create(:usergroup, :with_auditing) }
+      let (:child_usergroup) { FactoryBot.create(:usergroup) }
 
       test 'should audit when a child-usergroup is assigned to a parent-usergroup' do
-        recent_audit = @usergroup.audits.last
+        usergroup.usergroup_ids = [child_usergroup.id]
+        usergroup.save
```
Comment (ezr-ondrej) [inline test/models/usergroup_test.rb:322]: Theese two lines can still stay in the `setup` block.
Or rather in the `before` block (the same meaning, but sounds better IMHO).

File: test/models/usergroup_test.rb [hunk 2]
```diff
@@ -314,82 +314,88 @@ def populate_usergroups
...
-        @usergroup.role_ids = [@role.id]
-        @usergroup.save
-      end
+      let (:usergroup) { FactoryBot.create(:usergroup, :with_auditing) }
+      let (:role) { FactoryBot.create(:role) }
 
       test 'should audit when a role is assigned to a usergroup' do
-        recent_audit = @usergroup.audits.last
+        usergroup.role_ids = [role.id]
+        usergroup.save
```
Comment (ezr-ondrej) [inline test/models/usergroup_test.rb:350]: The same here.

**Review by ezr-ondrej — COMMENTED** [review]
Thanks @kgaikwad, awesome! I love the removal of the instance variables, just two nitpicks to stay DRY.

**Comment by kgaikwad** [comment]
Thank you @ekohl and @ezr-ondrej.
Updated the PR. Tested locally with SEED option, those tests are green. 
I hope it won't create any failures in the future as last time also tests were green on local and here in the job runs.

**Comment by ezr-ondrej** [comment]
[test katello]

**Comment by ezr-ondrej** [comment]
[test katello]

**Review by ezr-ondrej — APPROVED** [review]
Katello unrelated, even though it seems to insist on the failures.
Thanks @kgaikwad :+1:

**Pull request merged**